PluginsAndFeatures/azure-toolkit-for-intellij/rider/src/com/microsoft/intellij/runner/webapp/webappconfig/RiderWebAppRunState.kt

```diff
@@ -0,0 +1,441 @@
+
+            processHandler.setText(String.format(ZIP_FILE_CREATE_FOR_PROJECT, myModel.publishableProject?.projectName))
+            zipFile = zipProjectArtifacts(outDir, processHandler)
+
+            processHandler.setText(ZIP_DEPLOY_START_PUBLISHING)
+            publishZipToAzure(webAppName, zipFile, profile, processHandler)
+        } catch (e: Throwable) {
+            processHandler.setText("$DEPLOY_FAILED: $e")
+            throw e
+        } finally {
+            if (zipFile != null && zipFile.exists()) {
+                processHandler.setText(String.format(ZIP_FILE_DELETING, zipFile.path))
+                FileUtil.delete(zipFile)
+            }
+        }
+    }
+
+    /**
+     * Collect a selected project artifacts and get a [File] with out folder
+     * Note: For a DotNET Framework projects publish a copy of project folder as is
+     *       For a DotNet Core projects generate an publishable output by [MsBuildPublishingService]
+     *
+     * @param project IDEA [Project] instance
+     * @param publishableProject contains information about project to be published (isDotNetCore, path to project file)
+     *
+     * @return [File] to project content to be published
+     */
+    private fun collectProjectArtifacts(project: Project,
+                                        publishableProject: PublishableProjectModel,
+                                        processHandler: RunProcessHandler): File {
+
+        // Get out parameters
+        val publishService = MsBuildPublishingService.getInstance(project)
+        val (targetProperties, outPath) = publishService.getPublishToTempDirParameterAndPath()
+
+        val event = SyncEvent()
+        application.invokeLater {
+
+            val onFinish: (result: BuildResultKind) -> Unit = {
+                if (it != BuildResultKind.Successful) processHandler.setText(PROJECT_ARTIFACTS_COLLECTING_FAILED)
+                requestRunWindowFocus()
+                event.set()
+            }
+
+            if (publishableProject.isDotNetCore) {
+                publishService.invokeMsBuild(publishableProject.projectFilePath, listOf(targetProperties), false, false, onFinish)
+            } else {
+                publishService.webPublishToFileSystem(publishableProject.projectFilePath, outPath, false, false, onFinish)
+            }
+        }
+        event.wait(COLLECT_ARTIFACTS_TIMEOUT)
+
+        return outPath.toFile()
+    }
+
+    /**
+     * Archive a directory
+     *
+     * @param fromFile - file to be added to ZIP archive
+     * @param processHandler - a process handler to show a process message
+     * @param deleteOriginal - delete original file that is used for compression
+     *
+     * @return [File] Zip archive file
+     */
+    @Throws(IOException::class)
+    private fun zipProjectArtifacts(fromFile: File,
+                                    processHandler: RunProcessHandler,
+                                    deleteOriginal: Boolean = true): File {
```

Comment:
either clarify `deleteOriginal` or do not delete them on failing

Reply:
Delete on success.